(C++ SDK): Add sort documentation (#3095)
mongodb/docs-realm: source/sdk/cpp/crud/read.txt

.. literalinclude:: /examples/generated/cpp/crud.snippet.read-set.cpp
    :language: cpp

.. _cpp-sort-lists-and-results:

Sort Lists and Query Results
----------------------------

.. versionadded:: v0.5.0-preview

A sort operation allows you to configure the order in which Realm returns 
list objects and query results. You can sort based on one or more properties 
of the objects in the list or results collection. Realm only guarantees a 
consistent order if you explicitly sort.

Unlike using `std::sort <https://en.cppreference.com/w/cpp/algorithm/sort>`__, 
Realm's sort implementation preserves the lazy loading of objects. It does
not pull the entire set of results or list objects into memory, but only
loads them into memory when you access them.

To sort, call the ``.sort()`` function on a list or results set with one or more 
:cpp-sdk:`sort_descriptors <structrealm_1_1internal_1_1bridge_1_1sort__descriptor.html>`.

A ``sort_descriptor`` includes:

- The desired key path to sort by, as a string.

> - ``true`` or ``false`` to specify ascending or descending sort order. ``true``

Reviewer: suggestion
```suggestion
- A bool to specify sort order, where ``true``
```
